### ISSUE TITLE
TypeError: '<' not supported between instances of 'Candidate' and 'Candidate'  when running completion scenario

### ISSUE THREAD

**Issue by hlibbabii** [mast-group/OpenVocabCodeNLM#2]
4 MRR: 0.7272230155124293
Completion Length: 1256
5 MRR: 0.7315734824281155
Completion Length: 873
Traceback (most recent call last):
  File "/usr/src/app//OpenVocabCodeNLM/code_nlm.py", line 2315, in <module>
    tf.app.run()
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))
  File "/usr/src/app//OpenVocabCodeNLM/code_nlm.py", line 2224, in main
    FLAGS.dynamic, id_map, FLAGS.cache_ids, token_map)
  File "/usr/src/app//OpenVocabCodeNLM/code_nlm.py", line 1358, in completion
    (test_dataset.rev_vocab[id],))))
TypeError: '<' not supported between instances of 'Candidate' and 'Candidate'

**when running completion scenario:**

```
python code_nlm.py --completion True --data_path $DATA_HOME --train_dir $MODEL_DIR --test_filename $TEST_FILE --gru True --batch_size $BATCH_SIZE
```

**Comment by mpatsis** [comment]
As pointed by you and this stack overflow post:
https://stackoverflow.com/questions/43477958/typeerror-not-supported-between-instances-python
this seems to be happening when probabilities are equal because it cannot compare the candidates.

I've not managed to replicate the issue in python 2 though.

**Comment by mpatsis** [comment]
Fixed by commit: 808b0d190eb71a6de402ac58c3fce09e9ff81515
Thanks Hlib!
